__tests__/profile/charts/barchart.test.js

```diff
@@ -1,170 +1,329 @@
-import { configureBarchart } from '../../../src/js/profile/charts/barChart.js';
-
-import { parse, View }from 'vega';
+import {configureBarchart, configureBarchartDownload} from '../../../src/js/profile/charts/barChart.js';
+import {screen, fireEvent, getByText} from '@testing-library/dom'
+import {parse, View} from 'vega';
+import html from '../../../src/index.html';
+import {Chart} from '../../../src/js/profile/chart.js';
+import {Component} from '../../../src/js/utils.js'
 
 function renderVegaHeadless(spec) {
-  const runtime = parse(spec)
+    const runtime = parse(spec)
 
-  return new View(runtime);
+    return new View(runtime);
 }
 
 describe('configureBarchart', () => {
-  let data, metadata, config;
-  beforeEach(() => {
-    metadata = {
```

Comment:
We should try to have the demo data outside of the actual test js and import it when needed